### ISSUE TITLE
`UnicodeDecodeError` during initial setup of new package

### ISSUE THREAD

**Issue by aphearin** [astropy/package-template#265]
I just tried to start a new package using the `cookiecutter gh:astropy/package-template` and I get the following Traceback:

```
Traceback (most recent call last):
  File "/Users/aphearin/anaconda/bin/cookiecutter", line 6, in <module>
    sys.exit(cookiecutter.cli.main())
  File "/Users/aphearin/anaconda/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/aphearin/anaconda/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/aphearin/anaconda/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/aphearin/anaconda/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/aphearin/anaconda/lib/python2.7/site-packages/cookiecutter/cli.py", line 123, in main
    default_config=default_config,
  File "/Users/aphearin/anaconda/lib/python2.7/site-packages/cookiecutter/main.py", line 63, in cookiecutter
    no_input=no_input,
  File "/Users/aphearin/anaconda/lib/python2.7/site-packages/cookiecutter/repository.py", line 84, in determine_repo_dir
    no_input=no_input,
  File "/Users/aphearin/anaconda/lib/python2.7/site-packages/cookiecutter/vcs.py", line 132, in clone
    if 'not found' in clone_error.output.lower():
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 74: ordinal not in range(128)
```

My system info appears below:
```
               platform : osx-64
          conda version : 4.3.23
       conda is private : False
      conda-env version : 4.3.23
    conda-build version : 2.1.17
         python version : 2.7.13.final.0
       requests version : 2.18.4
       root environment : /Users/aphearin/anaconda  (writable)
    default environment : /Users/aphearin/anaconda/envs/py3_basic
       envs directories : /Users/aphearin/anaconda/envs
                          /Users/aphearin/.conda/envs
          package cache : /Users/aphearin/anaconda/pkgs
                          /Users/aphearin/.conda/pkgs
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/osx-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : /Users/aphearin/.condarc
             netrc file : None
           offline mode : False
             user-agent : conda/4.3.23 requests/2.18.4 CPython/2.7.13 Darwin/16.7.0 OSX/10.12.6    
                UID:GID : 501:20
```

**Comment by aphearin** [comment]
In case python 2 is the problem, I have tried again within a python 3 conda environment and I get the same error

**Comment by aphearin** [comment]
After rerunning once again, I can now successfully create a new package. This new run triggered the prompt, `You've cloned /Users/aphearin/.cookiecutters/package-template before. Is it okay to delete and re-clone it? [yes]:`. I chose `yes`and proceeded without problems. This message was not triggered following the commands that led to my raising this Issue. So whatever led to the exception seems to be resolved by whatever system config was done by creating the `.cookiecutters` file contents.

Not sure whether to close this now, by probably, since there is a simple workaround and the error was likely caused by local config issues.

**Comment by bsipocz** [comment]
@Cadair - could you have a look at this? is there anything we could document from the process, or do with the template to make this smoother?

**Comment by Cadair** [comment]
My first reaction on this is that it's a cookiecutter bug. 

However: `if 'not found' in clone_error.output.lower():` indicates it's something that's coming out of a failing clone.

Still, I think that the error is cookiecutter not handling it properly.

**Comment by aphearin** [comment]
I have not had any problems since then, and I've started other projects using the template since then. It's possible this was just an obscure user config issue. I'll revisit and see if I can reproduce it.

**Comment by aphearin** [comment]
Ok, I gave another try to making a new package and hit a different snag. 

```
$ cookiecutter gh:astropy/package-template

You've cloned /Users/aphearin/.cookiecutters/package-template before. 
Is it okay to delete and re-clone it? [yes]: 
A valid repository for "https://github.com/astropy/package-template.git" 
could not be found in the following locations:
/Users/aphearin/.cookiecutters/package-template

```

In case the `.cookiecutters` directory had become corrupted somehow, I blew it away and tried again but got the same message. 

I just checked `master` and noticed that development seems to be wandering ahead of the `cookiecutter` branch. Are y'all moving away from this way of doing things?

**Comment by Cadair** [comment]
This is probably because the default branch has swapped back to not-cookiecutter.

You need to specify `-b cookicutter`

On 4 October 2017 04:00:54 BST, Andrew Hearin <notifications@github.com> wrote:
>Ok, I gave another try to making a new package and hit a different
>snag. 
>
>```
>$ cookiecutter gh:astropy/package-template
>
>You've cloned /Users/aphearin/.cookiecutters/package-template before. 
>Is it okay to delete and re-clone it? [yes]: 
>A valid repository for
>"https://github.com/astropy/package-template.git" 
>could not be found in the following locations:
>/Users/aphearin/.cookiecutters/package-template
>
>```
>
>In case the `.cookiecutters` directory had become corrupted somehow, I
>blew it away and tried again but got the same message. 
>
>I just checked `master` and noticed that development seems to be
>wandering ahead of the `cookiecutter` branch. Are y'all moving away
>from this way of doing things?
>
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/astropy/package-template/issues/265#issuecomment-334037499

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by bsipocz** [comment]
Sorry about this @aphearin. The cookiecutter branch is turned out to be still experimental and need more docs before shipping it to the wider audience and we changed the default branch back to master.
Thus any feedback of using it is very valuable.

**Comment by aphearin** [comment]
Not a problem @bsipocz. I'm happy to provide feedback and also write some documentation. I built several different packages using the cookiecutter branch and was quite happy with the behavior. I would be perfectly happy to proceed if I were able to get `-b cookiecutter` to work as suggested by @Cadair, but this results in `Error: no such option: -b`.

**Comment by Cadair** [comment]
My mistake it's `-c cookiecutter`.

**Comment by Cadair** [comment]
This was fixed in audreyr/cookiecutter#905
